pkg/core/method.go

```diff
@@ -221,18 +221,29 @@ func (m *MethodCall) Execute(c CommunicationIntf, proto drive.SecurityProtocol,
 	// > response, any IF-RECV command for that ComID SHALL receive a ComPacket with a
 	// > Length field value of zero (no payload), an OutstandingData field value of 0x01, and a
 	// > MinTransfer field value of zero.
-	for i := 100; i >= 0; i-- {
-		resp, err = c.Receive(proto, ses)
-		if err != nil {
-			return nil, err
-		}
-		if len(resp) > 0 {
-			break
-		}
-		if i == 0 {
+	tmo := time.Duration(30 * time.Second)
+	if ses.ControlSession != nil &&
+		ses.ControlSession.TPerProperties.DefSessionTimeout != nil &&
+		int64(*ses.ControlSession.TPerProperties.DefSessionTimeout) > 0 {
+		tmo = time.Duration(int64(*ses.ControlSession.TPerProperties.DefSessionTimeout) * int64(time.Millisecond))
+	}
+	interval := time.Duration(50 * time.Millisecond)
+	ticker := time.NewTicker(interval)
+	timeoutC := time.After(tmo)
+	cont := true
+	for cont {
+		select {
+		case <-ticker.C:
+			resp, err = c.Receive(proto, ses)
+			if err != nil {
+				return nil, err
+			}
+			if len(resp) > 0 {
+				cont = false
+			}
+		case <-timeoutC:
```

Comment:
@yogev-lb just to understand -
the old code did have timeout (100 iterations X 10ms interval = 1s timeout). what you are saying is he default session timeout wasn't taken into account? 
can you explain the purpose of this timeout (better add such explanation to the commit message)?

Reply:
@adir-lb that is exectily what im saying
some operations - like revert for intel drives took around 5s
this caused a timeout.
the spec talks about a default timeout of 30s i think so i set it to this tmo value by default.